optimizely/helpers/condition.py

```diff
@@ -233,12 +240,205 @@ def substring_evaluator(self, index):
+
+    def semver_less_than_evaluator(self, index):
+        """ Evaluate the given semver less than match target version for the user version.
+
+      Args:
+        index: Index of the condition to be evaluated.
+
+      Returns:
+        Boolean:
+          - True if the user version is less than the target version.
+          - False if the user version is greater than or equal to the target version.
+        None:
+          - if the user version value is not string type or is null.
+    """
+        return self.compare_user_version_with_target_version(index) == -1
+
+    def semver_less_than_or_equal_evaluator(self, index):
+        """ Evaluate the given semver less than or equal to match target version for the user version.
+
+      Args:
+        index: Index of the condition to be evaluated.
+
+      Returns:
+        Boolean:
+          - True if the user version is less than or equal to the target version.
+          - False if the user version is greater than the target version.
+        None:
+          - if the user version value is not string type or is null.
+    """
+        return self.compare_user_version_with_target_version(index) <= 0
+
+    def semver_greater_than_or_equal_evaluator(self, index):
+        """ Evaluate the given semver greater than or equal to match target version for the user version.
+
+      Args:
+        index: Index of the condition to be evaluated.
+
+      Returns:
+        Boolean:
+          - True if the user version is greater than or equal to the target version.
+          - False if the user version is less than the target version.
+        None:
+          - if the user version value is not string type or is null.
+    """
+        return self.compare_user_version_with_target_version(index) >= 0
+
+    def split_semantic_version(self, target):
+        """ Method to split the given version.
+
+      Args:
+        target: Given version.
+
+      Returns:
+        List:
+          - The array of version split into smaller parts i.e major, minor, patch etc
+        Exception:
+          - if the given version is invalid in format
+    """
+        target_prefix = target
+        target_suffix = ""
+
+        if self.is_pre_release(target) or self.is_build(target):
+            target_parts = target.split(self.pre_release_separator() if self.is_pre_release(target) else self.build_separator())
+            if len(target_parts) < 1:
+                raise Exception(OptimizelyErrors.AttributeFormatInvalid)
+
+            target_prefix = str(target_parts[0])
+            target_suffix = target_parts[1:]
+
+        target_version_parts = target_prefix.split(".")
+        for part in target_version_parts:
+            if not part.isdigit():
+                raise Exception(OptimizelyErrors.AttributeFormatInvalid)
+
+        if target_suffix:
+            target_version_parts.extend(target_suffix)
+            return target_version_parts
+        else:
+            return target_version_parts
```

Comment:
when you are returning in both cases? Why not have a single return after if block ends.